smartsim/_core/launcher/util/shell.py

```diff
@@ -120,14 +122,19 @@ def execute_async_cmd(
         )
         time.sleep(0.2)
         popen_obj.poll()
+    except OSError as e:
+        raise ShellError("Failed to run command", cmd_list, details=e) from None
+
+    err_msg = ""
+    try:
         if not popen_obj.is_running() and popen_obj.returncode != 0:
             output, error = popen_obj.communicate()
-            err_msg = ""
             if output:
                 err_msg += output.decode("utf-8") + " "
             if error:
                 err_msg += error.decode("utf-8")
             raise ShellError("Command failed immediately", cmd_list, details=err_msg)
     except OSError as e:
-        raise ShellError("Failed to run command", cmd_list, details=e) from None
+        raise ShellError("Command failed", cmd_list, details=e) from None
```

Comment:
Changing the text of an error message could impact users who have scripts that parse output looking for specific messages

Reply:
This is a great thought. I'm not sure why I ended up looking at it, but I changed it specifically because [another error](https://github.com/CrayLabs/SmartSim/blob/6a27c451d93b5c304d8c2757bf9e4fc8fb13e384/smartsim/_core/launcher/util/shell.py#L126) has the same message. If two separate locations have the same message, I think it results in confusion as to the true source of the error.

I'm glad to change this back if you don't find that convincing.

Reply:
Exact duplication in error message text is bad, so changing seems good in that case

Reply:
I've reverted this. I think you're right that it's out of scope of passing the linter. If we need it, it's simple to do later, too